File: SOC/src/cc/game/soc/core/PlayerTemp.java
package cc.game.soc.core;

import java.util.Collection;
import java.util.List;

public class PlayerTemp extends Player {

    public PlayerTemp() {}
    
    PlayerTemp(int playerNum) {
        setPlayerNum(playerNum);
    }
    
	public PlayerTemp(Player p) {
		copyFrom(p);
	}

	@Override
	public MoveType chooseMove(SOC soc, Collection<MoveType> moves) {
		throw new AssertionError("This should never get called");
	}

	@Override
	public Integer chooseVertex(SOC soc, Collection<Integer> vertexIndices, VertexChoice mode, Integer knightToMove) {
        throw new AssertionError("This should never get called");
	}

	@Override
	public Integer chooseRoute(SOC soc, Collection<Integer> routeIndices, RouteChoice mode, Integer shipToMove) {
        throw new AssertionError("This should never get called");
	}

	@Override
	public RouteChoiceType chooseRouteType(SOC soc) {
        throw new AssertionError("This should never get called");
	}

	@Override
	public Integer chooseTile(SOC soc, Collection<Integer> tileIndices, TileChoice mode) {
        throw new AssertionError("This should never get called");
	}

	@Override
	public Trade chooseTradeOption(SOC soc, Collection<Trade> trades) {
        throw new AssertionError("This should never get called");
	}

	@Override
	public Integer choosePlayer(SOC soc, Collection<Integer> playerOptions, PlayerChoice mode) {
        throw new AssertionError("This should never get called");
	}

	@Override
	public Card chooseCard(SOC soc, Collection<Card> cards, CardChoice mode) {
        throw new AssertionError("This should never get called");
	}

	@Override
	public <T extends Enum<T>> T chooseEnum(SOC soc, EnumChoice mode, T [] values) {
        throw new AssertionError("This should never get called");
	}

	@Override
	public boolean setDice(SOC soc, List<Dice> die, int num) {
        throw new AssertionError("This should never get called");
	}

}

File: SOC/src/cc/game/soc/core/PlayerTemp.kt
package cc.game.soc.core

class PlayerTemp(playerNum: Int) : Player(playerNum) {

	constructor(p: Player) : this(0) {
		copyFrom(p)
	}

	override fun chooseMove(soc: SOC, moves: Collection<MoveType>): MoveType? {
		throw AssertionError("This should never get called")
	}

	override fun chooseVertex(soc: SOC, vertexIndices: Collection<Int>, mode: VertexChoice, knightToMove: Int?): Int? {
		throw AssertionError("This should never get called")
	}

	override fun chooseRoute(soc: SOC, routeIndices: Collection<Int>, mode: RouteChoice, shipToMove: Int?): Int? {
		throw AssertionError("This should never get called")
	}

	override fun chooseRouteType(soc: SOC): RouteChoiceType? {
		throw AssertionError("This should never get called")
	}

	override fun chooseTile(soc: SOC, tileIndices: Collection<Int>, mode: TileChoice): Int? {
		throw AssertionError("This should never get called")
	}

	override fun chooseTradeOption(soc: SOC, trades: Collection<Trade>): Trade? {
		throw AssertionError("This should never get called")
	}

	override fun choosePlayer(soc: SOC, playerOptions: Collection<Int>, mode: PlayerChoice): Int? {
		throw AssertionError("This should never get called")
	}

	override fun chooseCard(soc: SOC, cards: Collection<Card>, mode: CardChoice): Card? {
		throw AssertionError("This should never get called")
	}

	override fun <T : Enum<T>> chooseEnum(soc: SOC, mode: EnumChoice, values: Array<T>): T? {
		throw AssertionError("This should never get called")
	}

	override fun setDice(soc: SOC, die: List<Dice>, num: Int): Boolean {
		throw AssertionError("This should never get called")
	}
}